pkg/shared/util/grpc_utils.go

```diff
@@ -20,18 +20,77 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"strings"
 	"time"
 
+	"github.com/Masterminds/semver/v3"
 	"github.com/numaproj/numaflow-go/pkg/info"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/status"
 
+	"github.com/numaproj/numaflow"
 	sdkerr "github.com/numaproj/numaflow/pkg/sdkclient/error"
 	resolver "github.com/numaproj/numaflow/pkg/sdkclient/grpc_resolver"
 )
 
+func checkConstraint(version *semver.Version, constraint string) error {
+	c, err := semver.NewConstraint(constraint)
+	if err != nil {
+		return fmt.Errorf("error parsing constraint: %w", err)
+	}
+
+	isValid, _ := c.Validate(version)
+	if !isValid {
+		return fmt.Errorf("version did not meet constraint requirement")
```

Comment:
let's make the error message more useful by providing the content of `version` and `constraint`

Reply:
we provide their contents after the function call